### PR TITLE
Bump bundler version in sorbet_ruby

### PR DESCRIPTION
**Pull request by aisamanra**
### Motivation
Update the version of bundler we use in `sorbet_ruby`

### Test plan

See included automated tests. Tested along with the relevant changes in pay-server, c.f. [go/pr/617769](https://go/pr/617769).

### CODE REVIEW

**Comment by aisamanra** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_Neu7e5KrpJQp07
→ https://go/builds/bui_Neu7BW7Tirn5UW

**Comment by aisamanra** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_NeuEbt8yMQ4ATl
→ https://go/builds/bui_NeuEEMXTvaHTAB

**Comment by aisamanra** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_NpoxkOvbiPv69j
→ https://go/builds/bui_NpoxArSQOftqbf
→ https://go/builds/bui_Npox5eZNGVOqG7

File: third_party/ruby/build-ruby.bzl [hunk 1]
```diff
@@ -151,11 +151,6 @@ cp "$out_dir/bin/bundle" "$out_dir/bin/bundler"
 
 {install_gems}
 
-# Since we get our version of bundler from update_rubygems, we have to apply
-# this patch after everything is built. This is a bit of a hack, but the need
-# for the patch should go away once we upgrade Ruby and/or bundler anyway.
-{post_build_patch_command}
```
Comment (neilparikh) [inline third_party/ruby/build-ruby.bzl:157]: Should we also delete where `post_build_patch_command`/`post_build_patch_commands` is defined?

File: third_party/ruby_externals.bzl [hunk 2]
```diff
@@ -10,8 +10,8 @@ def register_ruby_dependencies():
 
     http_file(
         name = "rubygems_update_stripe",
-        urls = _rubygems_urls("rubygems-update-3.1.2.gem"),
-        sha256 = "7bfe4e5e274191e56da8d127c79df10d9120feb8650e4bad29238f4b2773a661",
+        urls = _rubygems_urls("rubygems-update-3.2.33.gem"),
```
Comment (neilparikh) [inline third_party/ruby_externals.bzl:13]: Does the `rubygems-update` gem contain `bundler`?

Comment (aisamanra) [reply third_party/ruby_externals.bzl:13]: Yes.

File: third_party/ruby/build-ruby.bzl (hunk 1 above)
Comment (aisamanra) [reply third_party/ruby/build-ruby.bzl:157]: I think that's a good idea, but I wanted to do that on a subsequent PR to minimize the diff.

**Review by neilparikh — APPROVED** [review]

**Pull request merged**